lib/profile.js

```diff
@@ -213,51 +224,65 @@ function enable2fa (args) {
       'Enabling two-factor authentication is an interactive opperation and ' +
       (conf.json ? 'JSON' : 'parseable') + 'output mode is not available'))
   }
-  log.notice('profile', 'Enabling two factor authentication for ' + mode)
-  const info = {
-    tfa: {
-      mode: mode
+
+  return Bluebird.try(() => {
+    // if they're using legacy auth currently then we have to update them to a
+    // bearer token before continuing.
+    if (conf.auth.basic) {
+      log.info('profile', 'Updating authentication to bearer token')
+      return profile.login(conf.auth.basic.username, conf.auth.basic.password, conf).then((result) => {
+        if (!result.token) throw new Error("Your registry " + conf.registry + "does not seem to support bearer tokens. Bearer tokens are required for two-factor authentication")
```

Comment:
I note that it's possible that they enabled TFA in another session, & the reason that they cannot authenticate here is because of their already-active TFA status. Does `profile.login` handle the OTP prompt in that case?

Reply:
yes

Reply:
ah, I see what you're saying… 

So this flow only happens if you run `npm profile enable-2fa`. So if you've `enabled-2fa` elsewhere but have a legacy auth on this account, I wouldn't expect you to be enabling 2fa here too? If you try you'll get an uncaught EOTP error. We _could_ catch it here, but it's really a more general problem because sudenly _every_ operation you do will ask for an OTP and I think it's far more likely that you'll get failures due to, for example, `npm install`.

Tweaking the error handler to be more specific, that is, to tell folks to `npm login` if they have bearer auth and just errored out due to EOTP (or E401 w/ no www-authenticate headers) would probably be good though.